opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -285,6 +286,9 @@ def update_name(self, name: str) -> None:
     def is_recording_events(self) -> bool:
         return True
 
+    def set_status(self, status: trace_api.Status) -> None:
+        self.status = status
```

Comment:
Please add the `has_ended` check like elsewhere (I wouldn't mind refactoring this in a common function while at it). I think it is especially valuable for the status, since it "describes the result of a finished span" so users could be confused into thinking they should set it after `end()`.